### PR TITLE
Cleaning up orders payloads

### PR DESCRIPTION
**Pull request by keith-miller**

### CODE REVIEW

File: api/order/definition.proto [hunk 1]
```diff
@@ -14,13 +14,11 @@ message Order {
...
   string total = 4;
-  string address = 6;
-  google.protobuf.Struct metadata = 7;
-  string request_ip = 8;
-  PaymentProviderId payment_provider_id = 9;
-  string item_types = 10;
-  string payment_provider_order_data = 11;
-  common.order.OrderState order_status = 12;
+  Address address = 5;
+  repeated PurchasedItem purchased_items = 6;
```
Comment (axcrest) [inline api/order/definition.proto:18]: For marketplace, we will only need a listing_id.  Do you want to add it now?  I believe the grpc can be oneof( purchaseditems, listing_id)

File: api/order/definition.proto [hunk 2]
```diff
@@ -14,13 +14,11 @@ message Order {
   string player_id = 2;
   string tax = 3;
   string total = 4;
-  string address = 6;
-  google.protobuf.Struct metadata = 7;
-  string request_ip = 8;
```
Comment (axcrest) [inline api/order/definition.proto:19]: still need request_ip right?

File: api/order/definition.proto [hunk 3]
```diff
@@ -14,13 +14,11 @@ message Order {
...
-  PaymentProviderId payment_provider_id = 9;
-  string item_types = 10;
-  string payment_provider_order_data = 11;
-  common.order.OrderState order_status = 12;
+  Address address = 5;
+  repeated PurchasedItem purchased_items = 6;
+  PaymentProviderId payment_provider_id = 7;
+  google.protobuf.Struct payment_provider_data = 8;
```
Comment (axcrest) [inline api/order/definition.proto:20]: We don't seem to need payment provider data.

File: api/order/definition.proto [hunk 4]
```diff
@@ -42,7 +40,7 @@ message PurchasedItem {
   string item_type_id = 3;
   string amount_paid = 4;
   string currency = 5;
```
Comment (axcrest) [inline api/order/definition.proto:42]: PurchasedItem as defined here assumes that description, render, memo, imgLarge, imgSmall, etc are set elsewhere?

File: api/order/definition.proto (hunk 2 above)
Comment (keith-miller) [reply api/order/definition.proto:19]: Hmm does the game send that now? Sorry, thought that was an internal thing lol. Will add back.

File: api/order/definition.proto (hunk 1 above)
Comment (keith-miller) [reply api/order/definition.proto:18]: Yes, let me add that now. Thanks

File: api/order/definition.proto (hunk 4 above)
Comment (keith-miller) [reply api/order/definition.proto:42]: Yes, set in the item type. We can grab that on the IVI side for them so this is easier to use.

File: api/order/definition.proto (hunk 2 above)
Comment (axcrest) [reply api/order/definition.proto:19]: It does send it, but backend doesn't use it atm.  Business side went back and forth between using billing address and IP.

File: api/order/definition.proto (hunk 4 above)
Comment (axcrest) [reply api/order/definition.proto:42]: K this change might need a little more discussion.  That data is stored in blockchain only (or metadata service?) so the ivi java code needs to look it up somewhere.  I agree this is an improvement vs making customers send it every time.  We just need to make sure those fields are relatively static.

Comment (keith-miller) [reply api/order/definition.proto:42]: Yeah, let's discuss.

File: api/order/definition.proto (hunk 3 above)
Comment (keith-miller) [reply api/order/definition.proto]: Fixed

File: api/order/definition.proto (hunk 2 above)
Comment (keith-miller) [reply api/order/definition.proto:19]: Then lets leave it off for now.

File: api/order/definition.proto (hunk 4 above)
Comment (keith-miller) [reply api/order/definition.proto:49]: Updated to make those params an optional pass in

**Review by axcrest — APPROVED** [review]

**Pull request merged**